subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/jar/SplitJarSnapshotCache.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.compile.incremental.jar;
+
+import com.google.common.collect.Maps;
+import org.gradle.api.internal.changedetection.state.WellKnownFileLocations;
+import org.gradle.internal.Factory;
+import org.gradle.internal.hash.HashCode;
+
+import java.io.File;
+import java.util.Map;
+
+public class SplitJarSnapshotCache implements JarSnapshotCache {
```

Comment:
`CombinedJarSnapshotCache`?

Reply:
I went with the existing name that we have in other places. The rationale being that it "splits" reads/writes between two levels.